src/Compiler/Driver/GraphChecking/DependencyResolution.fs

```diff
@@ -136,16 +120,16 @@ let filesInTrie (node: TrieNode) : Set<FileIndex> =
 /// - the namespace does not contain any children that can be referenced implicitly (eg. by type inference),
 /// then the main resolution algorithm does not create a link to any file defining the namespace.</para>
 /// <para>However, to satisfy the type-checker, the namespace must be resolved.
-/// This function returns a list of extra dependencies that makes sure that any such namespaces can be resolved (if it exists).
-/// For each unused open namespace we return one or more file links that define it.</para>
+/// This function returns an array with a potential extra dependencies that makes sure that any such namespaces can be resolved (if it exists).
+/// For each unused open namespace we might return one link that defined it.</para>
 /// </remarks>
 let collectGhostDependencies (fileIndex: FileIndex) (trie: TrieNode) (queryTrie: QueryTrie) (result: FileContentQueryState) =
-    // Go over all open namespaces, and assert all those links eventually went anywhere
+    // Go over all open namespaces, and assert all those links eventually went anywhere.
     Set.toArray result.OpenedNamespaces
-    |> Array.collect (fun path ->
+    |> Array.choose (fun path ->
```

Comment:
Do we need `Array.distinct` at the end or does that happen somewhere else?

Reply:
That happens later in https://github.com/dotnet/fsharp/blob/b47871a42ef61e6359bfc71635a231262e3a9328/src/Compiler/Driver/GraphChecking/DependencyResolution.fs#L245